go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1440,6 +1441,21 @@ func commonFlows(subnets []*net.IPNet, bridge *bridgeConfiguration) ([]string, e
 						defaultOpenFlowCookie, netConfig.ofPortPatch, bridgeMacAddress, ovnKubeNodeSNATMark,
 						config.Default.ConntrackZone, physicalIP.IP, netConfig.masqCTMark, ofPortPhys))
 
+				// table 0, packets coming from egressIP pods only from user defined networks. If an egressIP is assigned to
+				// this node, then all networks get a flow even if no pods on that network were selected for by this egressIP.
+				if util.IsNetworkSegmentationSupportEnabled() && config.OVNKubernetesFeature.EnableInterconnect &&
+					config.Gateway.Mode != config.GatewayModeDisabled && bridge.eipMarkIPs != nil {
+					if netConfig.masqCTMark != ctMarkOVN {
+						for mark, eip := range bridge.eipMarkIPs.GetIPv4() {
```

Comment:
This is really nice! having eipMarkIPs as source of truth to program SNAT OF rules.